app/src/main/java/org/lineageos/tv/launcher/MainActivity.kt

```diff
@@ -48,168 +53,112 @@ class MainActivity : FragmentActivity(R.layout.activity_main) {
     private val voiceAssistantButton by lazy { findViewById<ImageButton>(R.id.voice_assistant) }
 
     // Adapters
-    private val favoritesAdapter by lazy { FavoritesAdapter(this@MainActivity) }
-    private lateinit var mainVerticalAdapter: MainVerticalAdapter
-    private val appsAdapter by lazy { AppsAdapter(this@MainActivity) }
-
-    private lateinit var channels: List<PreviewChannel>
+    private val allAppsAdapter by lazy { AllAppsAdapter() }
+    private val favoritesAdapter by lazy { FavoritesAdapter() }
+    private val mainVerticalAdapter by lazy { MainVerticalAdapter() }
+    private val watchNextAdapter by lazy { WatchNextAdapter() }
 
-    private val packageReceiver by lazy { PackageReceiver() }
+    private val previewChannelAdapters = Collections.synchronizedMap(
```

Comment:
does this need to be sync?
I don't think we can get multiple collect at the same time.